tensorflow/lite/kernels/CMakeLists.txt

```diff
@@ -41,18 +41,37 @@ build_flatbuffers(
   ""
 )
 
+set(DELEGATE_PROVIDERS_SUPP
+  ${TFLITE_SOURCE_DIR}/delegates/external/external_delegate.cc
+  ${TFLITE_SOURCE_DIR}/nnapi/sl/SupportLibrary.cc
+  ${TFLITE_SOURCE_DIR}/tools/delegates/delegate_provider.cc
+  ${TFLITE_SOURCE_DIR}/tools/evaluation/utils.cc
+)
+
+set(DELEGATE_PROVIDERS
+  ${DELEGATE_PROVIDERS_SUPP}
+  ${TFLITE_SOURCE_DIR}/tools/delegates/coreml_delegate_provider.cc
```

Comment:
I think some delegate providers are only compilable on certain platforms, e.g., the CoreML delegate might be only compilable for iOS, and when using Bazel, we only add this dependency for iOS. So, I think we should not blindly add all providers here.

Reply:
right. Also NNAPI delegate should available only for Android

Reply:
Good point. Let me put it another way. The ([tensorflow/lite/CMakeLists.txt](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/CMakeLists.txt)) provides various _options_ using which the NNAPI, GPU and XNNPACK delegates can be enabled/disabled. As the user can toggle these _options_ even after the _configure_ phase is finished, it is required to have them in the _DELEGATE_PROVIDERS_ list. Conversely and admittedly, the delegates not mentioned in the root CMakeLists (Hexagon, External, GPU) are irrelevant and therefore were removed in the latest commit.